lib/misty/http/method_builder.rb

```diff
@@ -86,10 +86,16 @@ def prefixed_path(path)
 
       def query_param(data)
         if data.is_a? String
+          return '' if data == ''
           str = '?'
           str.force_encoding('ASCII-8BIT')
-          str << data unless data.empty?
+          str << data
+          return str
+        elsif data.is_a? Hash
+          return '' if data.empty?
+          return '?' + URI.encode_www_form(data)
         end
+        return nil
```

Comment:
I think there are too many `return` statements here, maybe `str = ''` at the beginning, alter str along the way and only `return str` at the end?